test/controllers/user.e2e-spec.ts

```diff
@@ -0,0 +1,44 @@
+import * as request from 'supertest';
+import { getManager } from 'typeorm';
+import { INestApplication } from '@nestjs/common';
+import { clearDB, createNestAppInstanceWithENVMock } from '../test.helper';
+import { Users } from 'src/entities/user.entity';
+
+describe('Users Controller', () => {
+  let app: INestApplication;
+  let mockConfig;
+
+  beforeEach(async () => {
+    await clearDB();
+  });
+
+  beforeAll(async () => {
+    ({ app, mockConfig } = await createNestAppInstanceWithENVMock());
+  });
+
+  describe('POST /api/users/register-user', () => { 
+    it ('should register a new user', async () => {
```

Comment:
Should we add a test for the invalid cases as well, like 

```js
it ('should return 422 when user email is already taken', async() => { ... })
it ('should return 422 when blank params are submitted', async() => { ... })
```

Reply:
We have a small problem here. The way that NestJS deals with 422 errors are with 400. Changing all of them to 422 is way too much work, meaning the resources I found to change them is almost minuscule or doing it manually, (changing every error to 422) which is something we don't want. A quick look at ToolJet; they seem to be ok with the idea of all these client side errors send off a 400 instead. My assumption as to why they did this is because NestJS is a pain in the a** when it comes to modifying these errors when we use their Validation pipe and Exception filters. Since everything is already built in, I suggest we let it be at 400 🤔

Reply:
![image](https://user-images.githubusercontent.com/28055566/167296489-4814f958-6817-4b4b-a218-bcb903e8ca6d.png)

Here are the new tests for testing exception messages 😸

Reply:
Nice! Looking great!